src/Command/CheckPasswordCommand.php

```diff
@@ -174,8 +174,15 @@ protected function execute(InputInterface $input, OutputInterface $output)
             $envVars['userdb_quota_rule'] = sprintf('*:storage=%dM', $user->getQuota());
         }
 
+        // Optionally create mail_crypt key pair and recovery token (when MAIL_CRYPT >= 3)
+        if ($this->mailCrypt >= 3 &&
+            false === $user->hasMailCrypt() &&
+            null === $user->getMailCryptPublicKey()) {
+            $this->mailCryptKeyHandler->create($user);
+        }
+
         // Optionally set mail_crypt environment variables for checkpassword-reply command
-        if (true === $this->mailCryptEnabled && true === $user->hasMailCrypt()) {
+        if ($this->mailCrypt >= 1 && $user->hasMailCrypt()) {
```

Comment:
You should add a constant for the MailCrypt Types

Reply:
Where would I define this constant? It needs to be available in `Command::CheckPasswordCommand`, `Command::MailCryptCommand`and `Handler::RegistrationHandler` at least.

Would you create a new class for global constants and import them with `use` in all required classes?

Unfortunately, `Handler::MailCryptKeyHandler` does not work here, as it's not imported by `Handler::RegistrationHandler`.